akka-http-core/src/test/scala/akka/http/impl/engine/client/ClientCancellationSpec.scala

```diff
@@ -4,41 +4,71 @@
 
 package akka.http.impl.engine.client
 
-import javax.net.ssl.SSLContext
+import akka.http.impl.util.{ AkkaSpecWithMaterializer, ExampleHttpContexts }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, headers }
+import akka.http.scaladsl.settings.{ ClientConnectionSettings, ConnectionPoolSettings }
+import akka.stream.scaladsl.{ Flow, Sink, Source }
+import akka.stream.testkit.{ TestPublisher, TestSubscriber, Utils }
 
 import scala.concurrent.Await
 import scala.concurrent.duration._
 
-import akka.http.impl.util.AkkaSpecWithMaterializer
-import akka.http.scaladsl.{ ConnectionContext, Http }
-import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
-import akka.stream.SystemMaterializer
-import akka.stream.scaladsl.{ Flow, Sink, Source }
-import akka.stream.testkit.{ TestPublisher, TestSubscriber, Utils }
-import akka.http.scaladsl.model.headers
-
 class ClientCancellationSpec extends AkkaSpecWithMaterializer {
-  // TODO document why this explicit materializer is needed here?
-  val noncheckedMaterializer = SystemMaterializer(system).materializer
-
   "Http client connections" must {
-    val address = Await.result(
+    "support cancellation in simple outgoing connection" in Utils.assertAllStagesStopped(new TestSetup {
+      testCase(
+        Http().outgoingConnection(address.getHostName, address.getPort))
+    })
+
+    "support cancellation in pooled outgoing connection" in Utils.assertAllStagesStopped(new TestSetup {
+      testCase(
+        Flow[HttpRequest]
+          .map((_, ()))
+          .via(Http().cachedHostConnectionPool(address.getHostName, address.getPort))
+          .map(_._1.get)
+      )
+    })
+
+    "support cancellation in simple outgoing connection with TLS" in Utils.assertAllStagesStopped(new TestSetup {
+      pending
```

Comment:
Didn't manage to fix this test here, since `Utils.assertAllStagesStopped` is completely broken with TlsModule in place, it was impossible for now to debug what's going on.